WooCommerce/WooCommerceTests/POS/Controllers/PointOfSaleItemsControllerTests.swift

```
@@ -198,7 +198,7 @@ final class PointOfSaleItemsControllerTests {
                                                   buttonText: "Retry")
 
         // When
-        await sut.loadNextItems()
+        try? await sut.loadNextItems()
```

Question: I'm not sure to understand why making the `try` optional actually makes the test pass. Shouldn't we'll be falling into the catch block in the following bit?
```
catch {
  itemsViewState = .init(containerState: .error(PointOfSaleErrorState.errorOnLoadingProducts()),
                                   itemsStack: ItemsStackState(root: .loaded(currentItems),
                                                               itemStates: currentItemStates))
            throw error
}
```
Which should set the state, then throw the error, and we would catch it in a test via:
```
        do {
            try await sut.loadNextItems()
        } catch {
            #expect(itemsViewState.containerState == .error(expectedError))
        }
``` 

I've tried this and it ofc it doesn't work: We do not fall into the catch block in the test target, so the expectation for error is never checked, I just don't understand why 😁  

Shouldn't the optional make it fail silently? Or that's precisely what happens? It fails silently but the `containerState` is already set to `PointOfSaleErrorState` so we just care about the state, not if we're throwing ?

---

I believe the optional try worked because `loadNextItems` triggers the container state to the error state synchronously, so we don't have to wait for the error to be thrown for the assertion. Moving it to do/catch as you suggested should also work, it passed for me locally and if I changed the assertion condition it failed so it was executed. do/catch looks nicer, less confusing, and Swift Testing doesn't require the XCTest extension workaround. I updated it in https://github.com/woocommerce/woocommerce-ios/pull/14833/commits/7849bbba5a8ce7e8c5c3e46e66d7d6ab4a792560.